jwst/assign_wcs/util.py

```diff
@@ -88,6 +89,70 @@ def _reproject(x, y):
     return _reproject
 
 
+def compute_scale_transform(refwcs, fiducial):
+    """Compute scaling transform
+    """
+    x0, y0 = refwcs.backward_transform(*fiducial)
+
+    x1 = x0 + 1
+    y1 = y0 + 1
+    ra0, dec0 = refwcs(x0, y0)
+    ra_xdir, dec_xdir = refwcs(x1, y0)
+    ra_ydir, dec_ydir = refwcs(x0, y1)
+
+    position0 = SkyCoord(ra=ra0, dec=dec0, unit='deg')
+    position_xdir = SkyCoord(ra=ra_xdir, dec=dec_xdir, unit='deg')
+    position_ydir = SkyCoord(ra=ra_ydir, dec=dec_ydir, unit='deg')
+
+    xscale = np.abs(position0.separation(position_xdir).value)
+    yscale = np.abs(position0.separation(position_ydir).value)
+    scale = np.sqrt(xscale * yscale)
+
+    return astmodels.Scale(scale) & astmodels.Scale(scale)
+
+
+def calc_rotation_matrix(angle, vparity=1):
+    """ Calculate the rotation matrix.
+
+    Parameters
+    ----------
+    angle : float in radians
+        The angle to create the matrix.
+
+    vparity : int
+        The x-axis parity, usually taken from
+        the JWST SIAF parameter VIdlParity.
+        Value should be "1" or "-1".
+
+    Returns
+    -------
+    matrix: [pc1_1, pc1_2, pc2_1, pc2_2]
+        The rotation matrix
+
+    Notes
+    -----
+    The rotation is
+
+       ----------------
+       | pc1_1  pc2_1 |
+       | pc1_2  pc2_2 |
+       ----------------
+
+    where:
+        pc1_1 = vparity * cos(angle)
+        pc1_2 = sin(angle)
+        pc2_1 = -1 * vparity * sin(angle)
+        pc2_2 = cos(angle)
+    """
```

Comment:
The order of the matrix in the comments is `pc11, pc21, ...`, and normally it's `pc11, pc12,...`.
It's probably best to remove the comments unless there's a special meaning. They are a copy of the code.